dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -719,6 +727,176 @@ public Limits setUserLimits(@ApiParam(hidden = true) @Auth User authUser,
+    public Set<String> getUserOrganizations(@ApiParam(hidden = true) @Auth User authUser,
+                                             @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @PathParam("gitRegistry") TokenType gitRegistry) {
+        User foundUser = userDAO.findById(authUser.getId());
+
+        List<Token> scTokens = tokenDAO.findByUserId(foundUser.getId())
+                .stream()
+                .filter(token -> Objects.equals(token.getTokenSource(), gitRegistry))
+                .collect(Collectors.toList());
+
+        if (scTokens.size() > 0) {
+            Token scToken = scTokens.get(0);
+            SourceCodeRepoInterface sourceCodeRepo = SourceCodeRepoFactory.createSourceCodeRepo(scToken, client);
+
+            final Map<String, String> repositoryUrlToName = sourceCodeRepo.getWorkflowGitUrl2RepositoryId();
+            Set<String> organizations = repositoryUrlToName.values().stream().map(repository -> repository.split("/")[0]).collect(Collectors.toSet());
+            return organizations;
+        }
+        return new HashSet<>();
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/registries/{gitRegistry}/organizations/{organization}")
+    @Operation(operationId = "getUserOrganizationRepositories")
+    @ApiOperation(nickname = "getUserOrganizationRepositories", value = "Get all of the repositories for an organization for a given git registry accessible to the logged in user.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = String.class, responseContainer = "Set")
+    public Set<String> getUserOrganizationRepositories(@ApiParam(hidden = true) @Auth User authUser,
+                                            @ApiParam(value = "Git registry", required = true, allowableValues = "GITHUB_COM, GITLAB_COM, BITBUCKET_ORG") @PathParam("gitRegistry") TokenType gitRegistry,
+                                            @ApiParam(value = "Git organization", required = true) @PathParam("organization") String organization) {
+        User foundUser = userDAO.findById(authUser.getId());
+
+        List<Token> scTokens = tokenDAO.findByUserId(foundUser.getId())
+                .stream()
+                .filter(token -> Objects.equals(token.getTokenSource(), gitRegistry))
+                .collect(Collectors.toList());
+
+        if (scTokens.size() > 0) {
+            Token scToken = scTokens.get(0);
+            SourceCodeRepoInterface sourceCodeRepo = SourceCodeRepoFactory.createSourceCodeRepo(scToken, client);
+
+            final Map<String, String> repositoryUrlToName = sourceCodeRepo.getWorkflowGitUrl2RepositoryId();
+            Set<String> repositories = repositoryUrlToName.values().stream().filter(repository -> repository.startsWith(organization + "/")).collect(Collectors.toSet());
```

Comment:
Up to this line, this method is exactly the same as the previous method... You could have a method that returns a `Map<String, String>` or an `Optional<Map<String,String>>`, and call it from both places.